crypto/txt_db/txt_db.c

```diff
@@ -21,9 +21,9 @@ TXT_DB *TXT_DB_read(BIO *in, int num)
 {
     TXT_DB *ret = NULL;
     int esc = 0;
-    int i, add, n;
-    int size = BUFSIZE;
-    int offset = 0;
+    int n;
+    size_t i, add, size = BUFSIZE, offset = 0;
+    size_t unum = num > 0 ? num : 0;
```

Comment:
!

Reply:
Wouldn't it just be better to return `NULL`? At the very least, what will happen with the `malloc()`s in lines 43 and 45 if `num < 0`?